### PR TITLE
[Helm charts] Add support to set resources for controller pods

### PR DESCRIPTION
**Pull request by guilhermeoki**
### Proposed changes
Add support to set resources for controller pods in helm charts.

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [X] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/master/CONTRIBUTING.md) doc
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have checked that all unit tests pass after adding my changes
- [X] I have updated necessary documentation
- [X] I have rebased my branch onto master
- [X] I will ensure my PR is targeting the master branch and pulling from my branch from my own fork

### CODE REVIEW

File: deployments/helm-chart/values.yaml [hunk 1]
```diff
@@ -74,6 +74,17 @@ controller:
   ## The termination grace period of the Ingress controller pod.
   terminationGracePeriodSeconds: 30
 
+  ## The resources of the Ingress controller pods.
+  resources: {}
+  # To specify resources, uncomment the following lines, adjust them as necessary,
+  # and remove the curly braces after 'resources:'.
+  #  limits:
```
Comment (Dean-Coakley) [inline deployments/helm-chart/values.yaml]: This indentation level is a bit inconsistent with what we already have in this file.
I'd prefer something like:
```yaml
  resources: {}
    # limits:
    #   cpu: 100m
    #   memory: 64Mi
    # requests:
    #   cpu: 100m
    #   memory: 64Mi
```

File: deployments/helm-chart/values.yaml [hunk 2]
```diff
@@ -74,6 +74,17 @@ controller:
   ## The termination grace period of the Ingress controller pod.
   terminationGracePeriodSeconds: 30
 
+  ## The resources of the Ingress controller pods.
+  resources: {}
+  # To specify resources, uncomment the following lines, adjust them as necessary,
```
Comment (Dean-Coakley) [inline deployments/helm-chart/values.yaml]: I'd prefer to remove these lines. (79-80)
I think it's reasonable to expect the user to know this and we don't have similar comments elsewhere.

File: deployments/helm-chart/values.yaml (hunk 1 above)
Comment (guilhermeoki) [reply deployments/helm-chart/values.yaml]: Done! I think the indentation is right now :)

File: deployments/helm-chart/templates/controller-daemonset.yaml [hunk 3]
```diff
@@ -67,6 +67,8 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: metadata.name
+        resources:
+{{ toYaml .Values.controller.resources | indent 12 }}
```
Comment (Dean-Coakley) [inline deployments/helm-chart/templates/controller-daemonset.yaml]: After https://github.com/nginxinc/kubernetes-ingress/pull/636#discussion_r307244549 this indentation needs to be adjusted
```suggestion
{{ toYaml .Values.controller.resources | indent 10 }}
```

File: deployments/helm-chart/templates/controller-deployment.yaml [hunk 4]
```diff
@@ -56,6 +56,8 @@ spec:
         - name: prometheus
           containerPort: {{ .Values.prometheus.port }}
 {{- end }}
+        resources:
+{{ toYaml .Values.controller.resources | indent 12 }}
```
Comment (Dean-Coakley) [inline deployments/helm-chart/templates/controller-deployment.yaml]: After https://github.com/nginxinc/kubernetes-ingress/pull/636#discussion_r307244549 this indentation needs to be adjusted
```suggestion
{{ toYaml .Values.controller.resources | indent 10 }}
```

Comment (guilhermeoki) [reply deployments/helm-chart/templates/controller-deployment.yaml]: Done! I have fixed the indentation.

File: deployments/helm-chart/templates/controller-daemonset.yaml (hunk 3 above)
Comment (guilhermeoki) [reply deployments/helm-chart/templates/controller-daemonset.yaml]: Done!

**Comment by Dean-Coakley** [comment]
Thanks a lot! @guilhermeoki

**Pull request merged**